nasl/nasl_builtin_find_service.c

```diff
@@ -1879,6 +1911,13 @@ plugin_do_run (struct script_infos *desc, GSList *h, int test_ssl)
                   else if (strncmp (line, "421", 3) == 0
                            && strstr (line, "smtp ") != NULL)
                     mark_smtp_server (desc, port, origline, trp);
+                  else if (line[0] != '\0' && (strstr (buffer, "mysql") != NULL)
```

Comment:
```suggestion
                  else if ((line[0] != '\0' || (strstr (buffer, "mysql") != NULL))
                           && (regex_match (
                                 buffer, "^.\0{3}\n[0-9.]+ [0-9a-z]+@[0-9a-z]+ release")
                               || regex_match (buffer,
                                               "^.\n{3}\n([0-9.]+)-(id([0-9]+)-)?release"
                                               " \\(([0-9a-z-]+)\\)")))
                    mark_sphinx (desc, port);
```

Those are a few binary responses i had collected:

```
0x00:  4B 00 00 00 0A 32 2E 38 2E 30 20 34 30 30 36 37    K....2.8.0 40067
0x10:  39 34 62 40 31 39 30 31 32 38 20 72 65 6C 65 61    94b@190128 relea
0x20:  73 65 00 01 00 00 00 01 02 03 04 05 06 07 08 00    se..............
0x30:  08 82 21 02 00 00 00 00 00 00 00 00 00 00 00 00    ..!.............
0x40:  00 00 01 02 03 04 05 06 07 08 09 0A 0B 0C 00       ...............
```

```
0x00:  61 00 00 00 0A 33 2E 30 2E 32 20 65 33 64 32 39    a....3.0.2 e3d29
0x10:  36 65 66 40 31 39 30 35 33 31 20 72 65 6C 65 61    6ef@190531 relea
0x20:  73 65 00 01 00 00 00 01 02 03 04 05 06 07 08 00    se..............
0x30:  08 82 21 02 00 08 00 15 00 00 00 00 00 00 00 00    ..!.............
0x40:  00 00 01 02 03 04 05 06 07 08 09 0A 0B 0C 00 6D    ...............m
0x50:  79 73 71 6C 5F 6E 61 74 69 76 65 5F 70 61 73 73    ysql_native_pass
0x60:  77 6F 72 64 00
```

```
0x00:  48 00 00 00 0A 32 2E 30 2E 38 2D 69 64 36 34 2D    H....2.0.8-id64-
0x10:  72 65 6C 65 61 73 65 20 28 72 33 38 33 31 29 00    release (r3831).
0x20:  01 00 00 00 01 02 03 04 05 06 07 08 00 08 82 21    ...............!
0x30:  02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 01    ................
0x40:  02 03 04 05 06 07 08 09 0A 0B 0C 0D                ............
```

Reply:
I only did "mysql" optional in the match. Other changes will not be committed because they produce false positive.

Reply:
Could you share some details on the false positives? The suggestion line anchors makes the regex even stricter so they should produce even less false detections.